src/main/java/org/ladocuploader/app/preparers/HouseholdDetailsPreparer.java

```diff
@@ -45,6 +50,23 @@ public Map<String, SubmissionField> prepareSubmissionFields(Submission submissio
 
         var uuid = householdMember.get("uuid");
         results.put("householdUSCitizen" + i, new SingleField("householdUSCitizenDerived", nonCitizens != null && nonCitizens.contains(uuid) ? "No" : "Yes", i + 1));
+
+        List<String> raceInput = (List)submission.getInputData().get(RACE_PREFIX + uuid + "[]");
```

Comment:
@bseeger we have a field called "permissionToAskAboutRace" that  determines whether the race questions were asked. If it's false then we wouldn't have displayed the race and ethnicity questions to the form filler.  

Maybe there is a benefit to using that here to exit out of the process sooner.

Reply:
Adding in this logic. Thanks for pointing that out, @analoo.